odoo/addons/base/models/res_partner.py

```diff
@@ -872,6 +872,17 @@ def write(self, vals):
         if 'is_company' in vals and not self.env.su and self.env.user.has_group('base.group_partner_manager'):
             result = super(ResPartner, self.sudo()).write({'is_company': vals.get('is_company')})
             del vals['is_company']
+
+        if vals.get('bank_ids'):
+            bank_ids = []
+            for bank_id in vals['bank_ids']:
+                if bank_id[0] == Command.SET:
+                    bank_ids.extend(bank_id[2])
+                else:
+                    bank_ids.append(bank_id[1])
```

Comment:
Only if it is `Command.LINK`, we don't want it for `Command.UNLINK` for instance